src/fsharp/TastOps.fs

```diff
@@ -4485,13 +4489,16 @@ and accFreeInExpr (opts: FreeVarOptions) x acc =
 and accFreeInExprLinear (opts: FreeVarOptions) x acc contf =   
     // for nested let-bindings, we need to continue after the whole let-binding is processed 
     match x with
-    | Expr.Let (bind, e, _, cache) -> 
-        let contf = contf << (fun free ->
-          unionFreeVars (freeVarsCacheCompute opts cache (fun () -> bindLhs opts bind (accBindRhs opts bind free))) acc )
-        accFreeInExprLinear opts e emptyFreeVars contf
+    | Expr.Let (bind, e, _, cache) ->
+        match tryGetFreeVarsCacheValue opts cache with
+        | ValueSome free -> contf (unionFreeVars free acc)
+        | _ ->
+            accFreeInExprLinear opts e emptyFreeVars (contf << (fun free ->
```

Comment:
Can we get a version without <<

Reply:
This is what we commonly do with CPS in the compiler. Personally, I think it's bit nicer than having to call `contf` with the result of the lambda's last expression.